src/FieldWarning/Assets/Units/UnitBehaviour.cs

```diff
@@ -19,13 +19,27 @@
 public abstract class UnitBehaviour : SelectableBehavior, Matchable<Vector3>
 {
     public const string UNIT_TAG = "Unit";
+    private const float ORIENTATION_RATE = 5.0f;
+    private const float TRANSLATION_RATE = 10.0f;
 
     public UnitData Data = UnitData.GenericUnit();
     public PlatoonBehaviour platoon { get; private set; }
     public bool IsAlive { get; private set; }
     public Pathfinder pathfinder { get; private set; }
     public AudioSource source { get; private set; }
 
+    // These are set by the subclass in DoMovement()
+    protected Vector3 position;
+    protected Vector3 rotation;
+
+    // Forward and right directions on the horizontal plane
+    protected Vector3 forward { get; private set; }
+    protected Vector3 right { get; private set; }
```

Comment:
I think this will eventually have to move into a component (`class MovementHandler` or similar), if we handle this with inheritance directly on UnitBehaviour it will quickly get bloated (imagine adding also state for armor, health, weapon and visibility directly under this class, ugh). In case you are curious, the ECS tutorial videos on the unity website happen to be based entirely on a movement example.

Reply:
VehicleBehaviour and UnitBehaviour are both already getting complicated, so st some point we will need to shift stuff to other classes for sure